File: sdks/java/src/main/java/com/jobcelis/JobcelisException.java
package com.jobcelis;

/**
 * Exception thrown when the Jobcelis API returns an error response.
 */
public class JobcelisException extends Exception {
    private final int statusCode;
    private final String detail;

    public JobcelisException(int statusCode, String detail) {
        super("Jobcelis API error " + statusCode + ": " + detail);
        this.statusCode = statusCode;
        this.detail = detail;
    }

    /** HTTP status code returned by the API. */
    public int getStatusCode() {
        return statusCode;
    }

    /** Error detail message from the API response. */
    public String getDetail() {
        return detail;
    }
}

File: sdks/kotlin/src/main/kotlin/com/jobcelis/JobcelisException.kt
package com.jobcelis

/**
 * Exception thrown when the Jobcelis API returns an error response.
 *
 * @property statusCode HTTP status code returned by the API.
 * @property detail Error detail message from the API response.
 */
class JobcelisException(
    val statusCode: Int,
    val detail: String
) : Exception("Jobcelis API error $statusCode: $detail")
